### ISSUE TITLE
🐛 [L1-001] Deprecated autocast import in evaluate.py

### ISSUE THREAD

**Issue by Zeesejo** [Zeesejo/medvision-ai#1]
## Bug Report — Level 1 (Atomic Bug)

**Hierarchy:** L1-001 → spawned by architecture-level import inconsistency

**File:** `src/evaluate.py` — Line 29

### Description
`evaluate.py` uses the old deprecated import path:
```python
from torch.cuda.amp import autocast
```
But `train.py` correctly uses the new path:
```python
from torch.amp import GradScaler, autocast
```
The `torch.cuda.amp.autocast` was deprecated in PyTorch 2.x and triggers a `DeprecationWarning`. On future PyTorch versions this will break.

### Fix
```python
# evaluate.py line 29 — replace:
from torch.cuda.amp import autocast
# with:
from torch.amp import autocast
```

**Labels:** `bug`, `low-priority`
**Severity:** Low — Warning now, breakage later
